crates/ott-balancer/src/balancer.rs

```diff
@@ -696,3 +696,87 @@ pub async fn dispatch_monolith_message(
 
     Ok(())
 }
+
+#[cfg(test)]
+mod test {
+    use std::net::Ipv4Addr;
+
+    use crate::discovery::{HostOrIp, MonolithConnectionConfig};
+
+    use super::*;
+
+    #[tokio::test]
+    async fn test_clients_add_remove() {
+        // a bunch of setup
+        let room_name = RoomName::from("test");
+        let mut ctx = BalancerContext::new();
```

Comment:
`BalancerContext` is simply a struct that describes the state of the balancer, including quantity of rooms per monolith, how many instances of a monolith per region, etc. correct?

Reply:
yeah